contracts/utils/deploy.js

```diff
@@ -32,6 +35,14 @@ function sleep(ms) {
 }
 
 const deployWithConfirmation = async (contractName, args, contract) => {
+  // check that upgrade doesn't corrupt the storage slots
+  await assertUpgradeIsSafe(hre, contractName);
+
+  // if upgrade happened on the mainnet save the new storage slot layout to the repo
```

Comment:
Rather than here. should this happen once the deploy is done and confirmed. Like after the call to withConfirmation below? So that in case of deploy error, the storage layout is not persisted.

Reply:
Yup makes a lot more sense, moving to there